### PR TITLE
Histogram followup

### LINKED ISSUE #4095: Histogram followup: Refine float/uint16 layer support
For float layers we'll need to refine the histogram support a bit. The backend should make a good guess at what the used float range is (by sampling) and send the minRange/maxRange values together with the histogram array data. Same for uint16 layers, probably increase the bucket size to 256.

### PR DESCRIPTION
**Pull request by youri-k**
- add sampling for float layers and better support for uint24 layers
### URL of deployed dev instance (used for testing):
- https://histogramfollowup.webknossos.xyz

### Steps to test:
- request histogram for `float` layer => response json should contain `min, max` entries
- request histogram for `uint24` layer => response json should contain 3 histograms for each channel

### Issues:
- fixes #4095 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [x] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

**Comment by youri-k** [comment]
@daniel-wer could you do the necessary frontend changes? Please note that the json keys slightly changed and that the backend now always sends an array of histograms.

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/FindDataService.scala [hunk 1]
```diff
@@ -231,28 +237,51 @@ class FindDataService @Inject()(dataServicesHolder: BinaryDataServiceHolder)(imp
...
         case floatData: Array[Float] =>
-          floatData.foreach(el => counts(Math.clamp(Math.roundDown(el * 255), 0, 255)) += 1)
+          val (min, max) = floatData.foldLeft((floatData(0), floatData(0))) {
+            case ((currMin, currMax), e) => (math.min(currMin, e), math.max(currMax, e))
+          }
+          val bucketSize = (max - min + 1) / 256
+          floatData.foreach(el => counts(Math.clamp(Math.roundDown((el - min) / bucketSize), 0, 255)) += 1)
+          extrema = (Some(min), Some(max))
+      }
+      if (!isUint24)
```
Comment (fm3) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/FindDataService.scala]: got me confused here for a moment, I’d suggest you use the positive condition and swap the statements :)

**Comment by daniel-wer** [comment]
@philippotto Frontend should be ready for review.
I added histogram support for uint16 and float layers, and fixed the uint24 histograms (they are colored now). (The min/max/minRange/maxRange naming is kind of confusing right now, maybe I should rework this code a bit.)

uint24_histogram

I updated the [data_types](https://github.com/scalableminds/webknossos-wiki/wiki/Test-Datasets#data_types-170-mb) Dataset to contain a properly scaled uint16 layer now (with values from 0 to `2**16` and not only `2**8`) and also another float layer that only contains values between 0 and 1. Please download the updated dataset to properly test this :)

@youri-k Could you also download the updated dataset and have a look at the `color_float_small` Layer. To me it looks like the min/max range is detected properly, but all the counts are in the first half then (they should be across the whole interval though, I think). Do you agree?

**Comment by MichaelBuessemeyer** [comment]
@philippotto It did not work on #4129 yet. Thanks for already solving this issue 🙂.

**Pull request merged**